packages/react-native-bpk-component-button/src/BpkButton.ios.js

```diff
@@ -20,100 +20,239 @@
+      ? {
+          backgroundColor: themeAttributes.buttonSecondaryBackgroundColor,
+          borderColor: themeAttributes.buttonSecondaryBorderColor,
+        }
+      : {
+          backgroundColor: backgroundColors.secondary,
+          borderColor: borderColors.secondary,
+        };
+  }
+  if (disabled) {
+    return {
+      backgroundColor: backgroundColors.disabled,
+      borderColor: borderColors.disabled,
+    };
+  }
+  return themeAttributes
+    ? {
+        backgroundColor: themeAttributes.buttonDestructiveBackgroundColor,
+        borderColor: themeAttributes.buttonDestructiveBorderColor,
+      }
+    : {
+        backgroundColor: backgroundColors.destructive,
+        borderColor: borderColors.destructive,
+      };
+};
+
+const getTextColor = (
+  type: ButtonType,
+  themeAttributes: ?Theme,
+  disabled: boolean,
+) => {
+  if (disabled) {
+    return textColors.disabled;
+  }
+  switch (type) {
+    case BUTTON_TYPES.primary:
+      return themeAttributes
+        ? themeAttributes.buttonPrimaryTextColor
+        : textColors.primary;
+    case BUTTON_TYPES.featured:
+      return themeAttributes
+        ? themeAttributes.buttonFeaturedTextColor
+        : textColors.featured;
+    case BUTTON_TYPES.secondary:
+      return themeAttributes
+        ? themeAttributes.buttonSecondaryTextColor
+        : textColors.secondary;
+    default:
+      return themeAttributes
+        ? themeAttributes.buttonDestructiveTextColor
+        : textColors.destructive;
+  }
+};
+
 const BpkButton = (props: Props) => {
   const {
     accessibilityLabel,
     disabled,
     icon,
+    iconAlignment,
     iconOnly,
     large,
-    onPress,
+    theme,
     title,
     type,
-    theme: themeProp,
     ...rest
   } = props;
 
-  if (!Object.values(BUTTON_TYPES).includes(type)) {
+  if (!Object.keys(BUTTON_TYPES).includes(type)) {
     throw new Error(
       `"${type}" is not a valid button type. Valid types are ${Object.keys(
         BUTTON_TYPES,
       ).join(', ')}`,
     );
   }
 
-  const shouldApplyTheme =
-    themeProp &&
-    (isTypeThemeable(type) && themeAttributesSupplied(type, themeProp));
-  const theme = shouldApplyTheme ? themeProp : null;
+  const themeAttributes = getThemeAttributes(
+    REQUIRED_THEME_ATTRIBUTES[type],
+    theme,
+  );
 
-  const accessibilityTraits = ['button'];
-  const gradientColors = getGradientColors(theme, props);
-  const buttonStyle = getStyleForElement('button', props);
-  const buttonTheme = getThemingForElement('button', theme, props);
+  const ButtonComponent = getButtonComponent(type);
+  const buttonColors = getButtonColors(type, themeAttributes, disabled);
+  const textColor = getTextColor(type, themeAttributes, disabled);
 
+  const accessibilityTraits = ['button'];
   if (disabled) {
     accessibilityTraits.push('disabled');
   }
 
+  // Icons can be passed in as a string or a BpkIcon. This normalises it
+  // so the inner component is always given a BpkIcon.
+  const iconFinal =
```

Comment:
nit: We can abstract this to a component too and use the same in both files

Reply:
Oh yeah great shout